### DISCUSSION TITLE
Encrypted columns

### DISCUSSION THREAD

**Discussion by runningjustin** [babyfish-ct/jimmer#602] [Ideas]
It would be nice to have a feature that automatically encrypts upon write and decrypts upon read. Maybe an annotation on the column in the Entity could trigger this behavior. An optional prefix would be good too so you know it's encrypted it's clear the value is encrypted to anybody looking at the raw text or even as a validation before auto-decrypting (e.g. `{enc}`)

**Comment by babyfish-ct** [comment]
can `ScalarProvider` resolve your problem?

**Reply by runningjustin** [reply]
Good suggestion - will give that a try.
